src/Glpi/Agent/Communication/AbstractRequest.php

```diff
@@ -207,6 +210,22 @@ public function handleHeaders()
      */
     public function handleRequest($data): bool
     {
+        $auth_required = \Config::getConfigurationValue('inventory', 'auth_required');
+        if ($auth_required === 'client_credentials') {
+            $request = new Request('POST', $_SERVER['REQUEST_URI'], $this->headers->getHeaders());
+            try {
+                $client = Server::validateAccessToken($request);
+                if (!\User::isNewID($client['user_id']) || !in_array('inventory', $client['scopes'], true)) {
```

Comment:
Could you explain why we check for `!\User::isNewID($client['user_id'])`? When I read this pice of code, I understand that we do NOT allow access to any valid user. A comment in the code would be appreciated.

Reply:
We ensure that the "user" is authenticated using `client_credentials` (and therefore there is no valid user ID associated with the client) and the `inventory` scope is used. If one or both conditions are not true, the request is denied.